src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,41 +1,122 @@
-const fetchFileExtension = /(.*)\.[^\.]*/;
-const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
-const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
-const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
-const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
-const fetchDoubleWhitespacesRegExp = /  */gm;
+import {fetchAllMatchesForAGroup} from "../regExpUtils";
+import builtInPredicates from "./builtInPredicates";
 
 export function removeFileExtension(path: string) {
+	const fetchFileExtension = /(.*)\.[^\.]*/;
```

Comment:
We could consider being extra nice to future developers and comment every regex with a comment
to this website https://regexper.com/ containing the regex in question in the link (after an #). For example here:
https://regexper.com/#/(.*)\.[^\.]*/

Reply:
Nah, if a developer wants to understand a regex further they'll do so. No need to add comments that will likely just desync at some point anyways.